model/riscv_sys_regs.sail

```diff
@@ -305,6 +305,43 @@ function in32BitMode() -> bool = {
   cur_architecture() == RV32
 }
 
+bitfield Seccfg : bits(64) = {
+  // Enable access to SEED in S-mode
+  SSEED  : 9,
+  // Enable access to SEED in U-mode
+  USEED  : 8,
+}
+
+function legalize_mseccfg(o : Seccfg, v : bits(64)) -> Seccfg = {
+  let v = Mk_Seccfg(v);
+  [
+    o with
+    SSEED = if currentlyEnabled(Ext_Zkr) then v[SSEED] else 0b0,
+    USEED = if currentlyEnabled(Ext_Zkr) then v[USEED] else 0b0,
+  ]
+}
+
+register mseccfg : Seccfg = legalize_mseccfg(Mk_Seccfg(zeros()), zeros())
+
+mapping clause csr_name_map = 0x747  <-> "mseccfg"
+mapping clause csr_name_map = 0x757  <-> "mseccfgh"
+
+// "mseccfg exists if Zkr is implemented, or if it is required by other processor features."
+function clause is_CSR_defined(0x747) = currentlyEnabled(Ext_Zkr) // mseccfg
+function clause is_CSR_defined(0x757) = currentlyEnabled(Ext_Zkr) & xlen == 32 // mseccfgh
+
+function clause read_CSR(0x747) = mseccfg.bits[xlen - 1 .. 0]
+function clause read_CSR(0x757 if xlen == 32) = mseccfg.bits[63 .. 32]
+
+function set_mseccfg(v : bits(64)) -> unit = {
+  mseccfg = legalize_mseccfg(mseccfg, v);
+}
+
+function clause write_CSR((0x747, value) if xlen == 32) = { set_mseccfg(mseccfg.bits[63 .. 32] @ value); mseccfg.bits[31 .. 0] }
```

Comment:
Other `write_CSR` clauses inline the `legalize_` call.  Perhaps we could do the same here for consistency?  It removes the need for a separate `set_` function.